tap-version-14-specification.md

```diff
@@ -0,0 +1,659 @@
+
+### Anything Else
+
+Any line that is not a valid version, plan, test point, yaml diagnostic,
+pragma, a blank line, or a bail out is invalid TAP.
+
+A Harness _may_ silently ignore invalid TAP lines, pass them through to its
+own stderr or stdout, or report them in some other fashion.  However, it
+_should not_ treat invalid TAP lines as a test failure by default.
+
+## EXAMPLES
+
+All names, places, and events depicted in any example are wholly fictitious
+and bear no resemblance to, connection with, or relation to any real
+entity. Any such similarity is purely coincidental, unintentional, and
+unintended.
+
+### Common with explanation
+
+The following TAP listing declares that six tests follow as well as
+provides handy feedback as to what the test is about to do. All six tests
+pass.
+
+```tap
+TAP version 14
+1..6
+#
+# Create a new Board and Tile, then place
+# the Tile onto the board.
+#
+ok 1 - The object isa Board
+ok 2 - Board size is zero
+ok 3 - The object isa Tile
+ok 4 - Get possible places to put the Tile
+ok 5 - Placing the tile produces no error
+ok 6 - Board size is 1
+```
+
+### Unknown amount and failures
+
+This hypothetical test program ensures that a handful of servers are online
+and network-accessible. Because it retrieves the hypothetical servers from
+a database, it doesn't know exactly how many servers it will need to ping.
+Thus, the test count is declared at the bottom after all the test points
+have run. Also, two of the tests fail. The YAML block following each
+failure gives additional information about the failure that may be
+displayed by the harness.
+
+```tap
+TAP version 14
+ok 1 - retrieving servers from the database
+# need to ping 6 servers
+ok 2 - pinged diamond
+ok 3 - pinged ruby
+not ok 4 - pinged saphire
+  ---
+  message: 'hostname "saphire" unknown'
+  severity: fail
+  ...
+ok 5 - pinged onyx
+not ok 6 - pinged quartz
+  ---
+  message: 'timeout'
+  severity: fail
+  ...
+ok 7 - pinged gold
+1..7
+```
+
+### Giving up
+
+This listing reports that a pile of tests are going to be run. However, the
+first test fails, reportedly because a connection to the database could not
+be established. The program decided that continuing was pointless and
+exited.
+
+```tap
+TAP version 14
+1..573
+not ok 1 - database handle
+Bail out! Couldn't connect to database.
+```
+
+### Skipping a few
+
+The following listing plans on running 5 tests. However, our program
+decided to not run tests 2 thru 5 at all. To properly report this, the
+tests are marked as being skipped.
+
+```tap
+TAP version 14
+1..5
+ok 1 - approved operating system
+# $^0 is solaris
+ok 2 - # SKIP no /sys directory
+ok 3 - # SKIP no /sys directory
+ok 4 - # SKIP no /sys directory
+ok 5 - # SKIP no /sys directory
+```
+
+### Skipping everything
+
+This listing shows that the entire listing is a skip. No tests were run.
+
+```tap
+TAP version 14
+1..0 # skip because English-to-French translator isn't installed
+```
+
+## Got spare tuits?
```

Comment:
I'm glad you're taking the time to give this such a careful read, it's very helpful having a non-native english speaker to find all the things that don't seem strange to me, since I've grown up with this strange language.

Reply:
>In english, sometimes you say that you'll do something "when you get around to it". That can be interpreted as "when you get a round tuit". No one knows what a "tuit" is, but if you find one that's round and doesn't have any corners, now you can do all those things you put off.

Aaaahhh!!! Thanks a lot for the explanation! And that's a good joke!

Reply:
Ha! I'm a native English speaker and I had no idea what tuit was. 😄  I assumed it was some Perl joke from years ago and rolled with it.

Reply:
> Ha! I'm a native English speaker and I had no idea what tuit was. smile I assumed it was some Perl joke from years ago and rolled with it.

I don't think Perl was the origin of it, but actually having round tuits made (wooden coins with tuit on it) was definitely a Perl joke from years ago 😋

Reply:
@Leont That joke predates Perl by at least 5 decades, maybe more.  My grandfather had a round ceramic disc with "tuit" written on it from some time around the great depression.  The height of humor in early 20th century America, apparently.

Reply:
ba460ac138568f2103cd09b1780acb98d15a82d7

Reply:
Are you calling us old-fashioned now? 😅